pkg/sql/resolver.go

```diff
@@ -406,11 +406,19 @@ func (p *planner) ResolveDescriptorForPrivilegeSpecifier(
 		if err != nil {
 			return nil, err
 		}
-		if err := validateColumnForHasPrivilegeSpecifier(
-			table,
-			specifier,
-		); err != nil {
-			return nil, err
+		if *specifier.IsSequence {
+			// has_table_privilege works with sequences, but has_sequence_privilege does not work with tables
+			if !table.IsSequence() {
+				return nil, pgerror.Newf(pgcode.WrongObjectType,
+					"\"%s\" is not a sequence", table.GetName())
```

Comment:
just an fyi: you can use the format verb `%q` and it will apply the quotes for you

Reply:
%q will also escape `"` characters which may not be what you want..., i prefer using `"%s" is not a sequence` instead.